script.js

```diff
@@ -1,12 +1,284 @@
-//You can edit ALL of the code here
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  // getALlEpisodesFromAPI() is an async function
+  // so it will produce a Promise
+  // Once it has fetched the data, then it will pass that data to makePageForEpisodes
+  let initialURL = "https://api.tvmaze.com/shows/82/episodes";
+
+  const allEpisodes =
+    getAllEpisodesFromAPI(initialURL).then(makePageForEpisodes);
+}
+
+async function getAllEpisodesFromAPI(url) {
+  let episodesStorage;
+
+  const response = await fetch(url);
+  episodesStorage = response.json();
+  return episodesStorage;
 }
 
 function makePageForEpisodes(episodeList) {
+  createHeader();
+  createShowsDropDownMenu();
+  loadNewShow();
+  createDropDownMenu(episodeList);
+  useDropdownToJumpToEpisode(episodeList);
+  createSearchbar();
+  createEpisodeCounter(episodeList);
+  createEpisodeCards(episodeList);
+  createFooter();
+}
+
+function createHeader() {
+  const rootElem = document.getElementById("root");
+  let pageHeaderElement = document.createElement("div");
+  pageHeaderElement.id = "page-header";
+  rootElem.append(pageHeaderElement);
+}
+
+function createShowsDropDownMenu() {
+  let dropDownMenuElement = document.createElement("select");
+  dropDownMenuElement.id = "show-select";
+  let defaultOption = document.createElement("option");
+  defaultOption.innerText = "Select Show";
+  dropDownMenuElement.append(defaultOption);
+
+  let allShows = getAllShows();
+
+  // good example of where to use a ternirary
+  allShows.sort(function (a, b) {
+    // if (a.name < b.name) {
+    //   return -1;
+    // }
+    // else {
+    //   return 1;
+    // }
+
+    return a.name < b.name ? 1 : -1;
```

Comment:
I think that the shows in your dropdown seem to appear in reverse order, so it may be that this ternary should have its possible values the other way round.

Reply:
Thanks for spotting this, Douglas! I accidentally left that in when I was at codebar. I've changed this back to an if statement as it's easier for me to understand.

Reply:
That sounds good! Definitely use the syntax that is easiest to understand!